src/core/lib/resource_quota/resource_quota.h

```diff
@@ -24,21 +24,29 @@ namespace grpc_core {
 
 class ResourceQuota : public RefCounted<ResourceQuota> {
  public:
-  ResourceQuota();
+  explicit ResourceQuota(std::string name);
   ~ResourceQuota() override;
 
   ResourceQuota(const ResourceQuota&) = delete;
   ResourceQuota& operator=(const ResourceQuota&) = delete;
 
-  std::shared_ptr<MemoryQuota> memory_quota() { return memory_quota_; }
+  MemoryQuotaPtr memory_quota() { return memory_quota_; }
 
   const RefCountedPtr<ThreadQuota>& thread_quota() { return thread_quota_; }
 
  private:
-  std::shared_ptr<MemoryQuota> memory_quota_;
+  MemoryQuotaPtr memory_quota_;
   RefCountedPtr<ThreadQuota> thread_quota_;
 };
 
+using ResourceQuotaPtr = RefCountedPtr<ResourceQuota>;
```

Comment:
I disagree with your preference here - since we've got public and private api's mixed (and need to), there's a mix of RefCountedPtr and std::shared_ptr, and typing it out became strictly more confusing - forcing users to remember which part was declared where to figure out what kind of pointer they needed.

Reply:
I think resolved.